### PR TITLE
[WIP] [FG:InPlacePodVerticalScaling] Pass only admitted resize request to pod workers

### LINKED ISSUE #125205: [FG:InPlacePodVerticalScaling] Slow reconcile when quickly reverting resize patch
### What happened?

While working on https://github.com/kubernetes/kubernetes/pull/125202 and testing the second case (patching a pod to perform an in-place resize and then quickly reverting the patch before the resize has been actuated), I've discovered some unexpected behavior. In this case the pod eventually reconciles but it takes about 3 minutes, with the following test output:

```
[sig-node] Pod InPlace Resize Container [Feature:InPlacePodVerticalScaling] Burstable QoS pod, three containers - no change for c1, increase c2 resources, decrease c3 (net decrease for pod) [sig-node, Feature:InPlacePodVerticalScaling]
k8s.io/kubernetes/test/e2e/node/pod_resize.go:1281
  STEP: Creating a kubernetes client @ 05/29/24 18:13:28.391
  I0529 18:13:28.391963 3291 util.go:499] >>> kubeConfig: /root/kind-test-config
  I0529 18:13:28.393961 3291 util.go:508] >>> kubeContext: kind-kind
  STEP: Building a namespace api object, basename pod-resize @ 05/29/24 18:13:28.394
  STEP: Waiting for a default service account to be provisioned in namespace @ 05/29/24 18:13:28.404
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 05/29/24 18:13:28.406
  STEP: Creating a kubernetes client @ 05/29/24 18:13:28.408
  I0529 18:13:28.408474 3291 util.go:499] >>> kubeConfig: /root/kind-test-config
  I0529 18:13:28.409838 3291 util.go:508] >>> kubeContext: kind-kind
  STEP: Building a namespace api object, basename pod-resize-resource-quota @ 05/29/24 18:13:28.41
  STEP: Waiting for a default service account to be provisioned in namespace @ 05/29/24 18:13:28.416
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 05/29/24 18:13:28.417
  STEP: Creating a kubernetes client @ 05/29/24 18:13:28.419
  I0529 18:13:28.419504 3291 util.go:499] >>> kubeConfig: /root/kind-test-config
  I0529 18:13:28.420717 3291 util.go:508] >>> kubeContext: kind-kind
  STEP: Building a namespace api object, basename pod-resize-errors @ 05/29/24 18:13:28.42
  STEP: Waiting for a default service account to be provisioned in namespace @ 05/29/24 18:13:28.427                                                                                                                                                                                                                                                                        
  STEP: Waiting for kube-root-ca.crt to be provisioned in namespace @ 05/29/24 18:13:28.428                                                                                                                                                                                                                                                                                 
  STEP: creating pod @ 05/29/24 18:13:28.429                                                                                                                                                                                                                                                                                                                                
  STEP: verifying the pod is in kubernetes @ 05/29/24 18:13:30.448                                                                                                                                                                                                                                                                                                          
  STEP: verifying initial pod resources, allocations, and policy are as expected @ 05/29/24 18:13:30.453                                                                                                                                                                                                                                                                    
  STEP: verifying initial pod status resources and cgroup config are as expected @ 05/29/24 18:13:30.453                                                                                                                                                                                                                                                                    
  STEP: patching pod for resize @ 05/29/24 18:13:30.457                                                                                                                                                                                                                                                                                                                     
  STEP: verifying pod patched for resize @ 05/29/24 18:13:30.47                                                                                                                                                                                                                                                                                                             
  STEP: patching pod for rollback @ 05/29/24 18:13:30.516                                                                                                                                                                                                                                                                                                                   
  STEP: verifying pod patched for rollback @ 05/29/24 18:13:30.529                                                                                                                                                                                                                                                                                                          
  STEP: waiting for rollback to be actuated @ 05/29/24 18:13:30.529                                                                                                                                                                                                                                                                                                         
  I0529 18:13:30.534186 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 -- 
 kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod -- ls 
  /sys/fs/cgroup/cgroup.controllers'
  I0529 18:13:30.679258 3291 builder.go:146] stderr: "Defaulted container \"c1\" out of: c1, c2, c3\n"
  I0529 18:13:30.679299 3291 builder.go:147] stdout: "/sys/fs/cgroup/cgroup.controllers\n"
  I0529 18:13:30.679331 3291 pod_resize.go:379] Namespace pod-resize-4426 Pod testpod Container c1 - looking for cgroup value 209715200 in path /sys/fs/cgroup/memory.max
  I0529 18:13:30.679379 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c1 -- head -n 1 /sys/fs/cgroup/memory.max'
  I0529 18:13:30.819936 3291 builder.go:146] stderr: ""
  I0529 18:13:30.820015 3291 builder.go:147] stdout: "209715200\n"
  I0529 18:13:30.820050 3291 pod_resize.go:379] Namespace pod-resize-4426 Pod testpod Container c1 - looking for cgroup value 20000 100000 in path /sys/fs/cgroup/cpu.max
  I0529 18:13:30.820117 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c1 -- head -n 1 /sys/fs/cgroup/cpu.max'
  I0529 18:13:30.907604 3291 builder.go:146] stderr: ""
  I0529 18:13:30.907650 3291 builder.go:147] stdout: "20000 100000\n"
  I0529 18:13:30.907669 3291 pod_resize.go:379] Namespace pod-resize-4426 Pod testpod Container c1 - looking for cgroup value 4 in path /sys/fs/cgroup/cpu.weight
  I0529 18:13:30.907714 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c1 -- head -n 1 /sys/fs/cgroup/cpu.weight'
  I0529 18:13:30.993819 3291 builder.go:146] stderr: ""
  I0529 18:13:30.993869 3291 builder.go:147] stdout: "4\n"
  I0529 18:13:30.993895 3291 pod_resize.go:379] Namespace pod-resize-4426 Pod testpod Container c2 - looking for cgroup value 314572800 in path /sys/fs/cgroup/memory.max
  I0529 18:13:30.993951 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c2 -- head -n 1 /sys/fs/cgroup/memory.max'
  I0529 18:13:31.078830 3291 builder.go:146] stderr: ""
  I0529 18:13:31.078876 3291 builder.go:147] stdout: "367001600\n"
  I0529 18:13:33.080354 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c2 -- head -n 1 /sys/fs/cgroup/memory.max'
  I0529 18:13:33.205904 3291 builder.go:146] stderr: ""
  I0529 18:13:33.205947 3291 builder.go:147] stdout: "367001600\n"
  I0529 18:13:35.207569 3291 builder.go:121] Running '/usr/local/bin/kubectl --server=https://127.0.0.1:60962 --kubeconfig=/root/kind-test-config --context=kind-kind --namespace=pod-resize-4426 exec testpod --namespace=pod-resize-4426 --container=c2 -- head -n 1 /sys/fs/cgroup/memory.max'
  I0529 18:13:35.335993 3291 builder.go:146] stderr: ""
  I0529 18:13:35.336037 3291 builder.go:147] stdout: "367001600\n"
```

With the last couple of lines continuing indefinitely until it reads the "rolled back" value from `memory.max` in `c2` (3 mins in my case). This also happens inconsistently, you may have to rerun the test case a couple of times to hit it. You can also run the full suite (`-ginkgo.focus="Feature:InPlacePodVerticalScaling"`) to hit it instead of the specific case provided below.


### What did you expect to happen?

After patching forwards and patching backwards, the pod should reach its initial state, and `memory.max` in `c2` should be set to its initial value.

### How can we reproduce it (as minimally and precisely as possible)?

- Check out https://github.com/kubernetes/kubernetes/pull/125202
- Uncomment [this line](https://github.com/kubernetes/kubernetes/pull/125202/files#diff-44a1da1c31a8cd8913e1b8cfb2893c5436bb7c60a4d51c9707f9368bf755ef8fR1391) - `patchAndVerifyAborted`
- Comment both `patchAndVerify` lines (so we only run the "aborted" case), and comment [this line](https://github.com/kubernetes/kubernetes/pull/125202/files#diff-44a1da1c31a8cd8913e1b8cfb2893c5436bb7c60a4d51c9707f9368bf755ef8fR1362)
- Run `e2e.test` with `-ginkgo.focus="Burstable QoS pod, three containers - decrease c1 resources" -num-nodes 2` (you can also run the full suite (`-ginkgo.focus="Feature:InPlacePodVerticalScaling"`) - you may have to run it a couple of times)

### Anything else we need to know?

This could also be an issue with my test changes, so please flag anything that seems suspicious. If you leave `patchAndVerify` uncommented (so patch -> wait for resize -> patch back -> wait for resize) and then run `patchAndVerifyAborted` (on the same case), you will hit a different bug, which I will file separately. 

### Kubernetes version

<details>

```console
$ kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"25", GitVersion:"v1.25.4", GitCommit:"872a965c6c6526caa949f0c6ac028ef7aff3fb78", GitTreeState:"clean", BuildDate:"2022-11-09T13:36:36Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v4.5.7
Server Version: version.Info{Major:"1", Minor:"31+", GitVersion:"v1.31.0-alpha.0.947+1ff1207d22ab5c-dirty", GitCommit:"1ff1207d22ab5cf442c8dafdf5bded1e32519873", GitTreeState:"dirty", BuildDate:"2024-05-28T19:28:33Z", GoVersion:"go1.22.3", Compiler:"gc", Platform:"linux/arm64"}
WARNING: version difference between client (1.25) and server (1.31) exceeds the supported minor version skew of +/-1
```

</details>


### Cloud provider

<details>
None
</details>


### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
$ uname -a
Linux docker-desktop 5.15.49-linuxkit #1 SMP PREEMPT Tue Sep 13 07:51:32 UTC 2022 aarch64 GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>


### Install tools

<details>
kind version 0.17.0
</details>


### Container runtime (CRI) and version (if applicable)

<details>

```console
NAME                 STATUS   ROLES           AGE     VERSION                                    INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION     CONTAINER-RUNTIME
kind-control-plane   Ready    control-plane   6h29m   v1.31.0-alpha.0.947+1ff1207d22ab5c-dirty   172.18.0.4    <none>        Ubuntu 22.04.1 LTS   5.15.49-linuxkit   containerd://1.6.9
kind-worker          Ready    <none>          6h29m   v1.31.0-alpha.0.947+1ff1207d22ab5c-dirty   172.18.0.2    <none>        Ubuntu 22.04.1 LTS   5.15.49-linuxkit   containerd://1.6.9
kind-worker2         Ready    <none>          6h29m   v1.31.0-alpha.0.947+1ff1207d22ab5c-dirty   172.18.0.3    <none>        Ubuntu 22.04.1 LTS   5.15.49-linuxkit   containerd://1.6.9
```

</details>


### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### LINKED ISSUE #126033: [FG:InPlacePodVerticalScaling] Infeasible resize is actuated by restarting container
### What happened?

If a container is restarted along with an exit while its resizing is stuck in `Infeasible` or `Deferred`, the unacceptable resizing is actuated:
1. Request an infeasible resize (exceeds node allocatable cpu):
    ```
   $ kubectl patch pod resize-pod --patch '{"spec":{"containers":[{"name":"resize-container", "resources":{"requests":{"cpu":"100"}, "limits":{"cpu":"100"}}}]}}'
   pod/resize-pod patched

    ```
2. Kill the container manually (`restartPolicy` is `Always`):
   ```
   $ kubectl exec resize-pod -- kill 1
   ```
3. Then, it looks still stuck in `Infeasible` or `Deferred` on the API pod status. However, the cgroup of the container is configured as requested:
   ```
   $ kubectl get pod resize-pod -o jsonpath='{.status.resize}'
   Infeasible
   ```
   ```
   $ kubectl exec resize-pod -- cat /sys/fs/cgroup/cpu.max
   10000000 100000
   # 10000000 = newLimit(100) * 1000mCPU * period(100000) / 1000mCPU
   ```
   ```
   $ kubectl exec resize-pod -- cat /sys/fs/cgroup/cpu.weight
   3906
   # 3906 = ((newRequest(100) * 1000mCPU * SharesPerCPU(1024) / 1000mCPU) - 2 ) * 9999 / 262142
   ```

However, it does not seem that this issue affects actual resource consumption. Because the pod cgroup is not updated in this case, the container resource will keep limited. In addition, since `AllocatedResources` in the container is not updated, this infeasible resizing will not affect the pod resource calculation of the scheduler.

### What did you expect to happen?

This killed container is restarted with the old spec.

### How can we reproduce it (as minimally and precisely as possible)?

1. Enable `InPlacePodVerticalScaling`.
2. Create a pod whose restartPolicy is `Always`:
    <details>

   ```
   apiVersion: v1
   kind: Pod
   metadata:
     creationTimestamp: null
     labels:
       run: resize-pod
     name: resize-pod
   spec:
     containers:
     - image: busybox
       name: resize-container
       command:
         - sh
         - -c
         - trap "exit 0" SIGTERM; while true; do sleep 1; done
       resources:
         requests:
           cpu: 200m
           memory: 200Mi
         limits:
           cpu: 200m
           memory: 200Mi
       resizePolicy:
       - resourceName: cpu
         restartPolicy: NotRequired
       - resourceName: memory
         restartPolicy: NotRequired
     restartPolicy: Always
   ```

    </details>
3. Request an infeasible resize:
   ```
   kubectl patch pod resize-pod --patch '{"spec":{"containers":[{"name":"resize-container", "resources":{"requests":{"cpu":"100"}, "limits":{"cpu":"100"}}}]}}'
   ```
4. Kill the container:
   ```
   kubectl exec resize-pod -- kill 1
   ```

   

### Anything else we need to know?

By executing `kubectl get --watch`, we can see the resized status in a moment:
```
$ kubectl get pod resize-pod -o jsonpath='spec: {.spec.containers[0].resources}{"\nallocatedResources: "}{.status.containerStatuses[0].allocatedResources}{"\nstatus: "}{.status.containerStatuses[0].resources}{"\nresize: "}{.status.resize}{"\n"}' -w
spec: {"limits":{"cpu":"100","memory":"200Mi"},"requests":{"cpu":"100","memory":"200Mi"}}
allocatedResources: {"cpu":"200m","memory":"200Mi"}
status: {"limits":{"cpu":"100","memory":"200Mi"},"requests":{"cpu":"100","memory":"200Mi"}}  # <- See this line
resize: 
spec: {"limits":{"cpu":"100","memory":"200Mi"},"requests":{"cpu":"100","memory":"200Mi"}}
allocatedResources: {"cpu":"200m","memory":"200Mi"}
status: {"limits":{"cpu":"200m","memory":"200Mi"},"requests":{"cpu":"200m","memory":"200Mi"}}
resize: Infeasible
```

The resized pod status is set here:
https://github.com/kubernetes/kubernetes/blob/37f733a657ef71d66177d00f9b7d47ec507dedd3/pkg/kubelet/kubelet.go#L1802
However, the pod status is overwritten with an old pod status immediately here:
https://github.com/kubernetes/kubernetes/blob/37f733a657ef71d66177d00f9b7d47ec507dedd3/pkg/kubelet/kubelet.go#L2820

This issue is mentioned in another issue: https://github.com/kubernetes/kubernetes/issues/125205#issuecomment-2143532151

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: v1.29.6
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.0
```
</details>

### Cloud provider

N/A

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### LINKED ISSUE #126527: [FG:InPlacePodVerticalScaling] Handle pod resize even if the pod has not started yet
### What happened?

If an unacceptable pod resizing that causes `Deferred` or `Infeasible` is requested before the container is started (for example, while an init container is running), the container is started with the unacceptable spec.

```
$ kubectl create -f pod.yaml; sleep 5; kubectl patch pod resize-pod --patch '{"spec": {"containers": [{"name": "resize-container", "resources":{"requests": {"cpu": "100"}, "limits": {"cpu": "100"}}}]}}'
pod/resize-pod created
pod/resize-pod patched
$ kubectl get pod resize-pod -o jsonpath='spec: {.spec.containers[0].resources}{"\nallocatedResources: "}{.status.containerStatuses[0].allocatedResources}{"\nstatus: "}{.status.containerStatuses[0].resources}{"\nresize: "}{.status.resize}{"\n"}'
spec: {"limits":{"cpu":"100","memory":"200Mi"},"requests":{"cpu":"100","memory":"200Mi"}}
allocatedResources: {"cpu":"200m","memory":"200Mi"}
status: {"limits":{"cpu":"100","memory":"200Mi"},"requests":{"cpu":"100","memory":"200Mi"}}
resize: Infeasible
```

The pod is admitted with the initial spec when the pod is created. Then, the resized spec is not verified for admission because the pod is not running yet:
https://github.com/kubernetes/kubernetes/blob/dbc2b0a5c7acc349ea71a14e49913661eaf708d2/pkg/kubelet/kubelet.go#L2811-L2814
As a result, the container is started with the unacceptable spec. Eventually, the pod gets into `Infeasible` resize status after the pod is started because the allocated resources that are not updated differs from the resized pod spec.

It does not seems that this issue affects actual resource consumption similarly to #126033. Because the pod cgroup is not updated in this case, the container resource will keep limited. In addition, since `AllocatedResources` in the container is not updated, this infeasible resizing will not affect the pod resource calculation of the scheduler.

### What did you expect to happen?

The pod is started with the initial spec and gets into `Infeasible` resize status or the pod fails to start.

### How can we reproduce it (as minimally and precisely as possible)?

0. Enable `InPlacePodVerticalScaling`.
1. Create a pod with an init container that takes a few seconds to complete:
    <Details>

   ```
   apiVersion: v1
   kind: Pod
   metadata:
     creationTimestamp: null
     labels:
       run: resize-pod
     name: resize-pod
   spec:
     initContainers:
     - image: busybox
       name: init-container
       command:
         - sleep
         - "10"
       resources:
         requests:
           cpu: 100m
           memory: 100Mi
         limits:
           cpu: 100m
           memory: 100Mi
     containers:
     - image: busybox
       name: resize-container
       command:
         - sh
         - -c
         - trap "exit 0" SIGTERM; while true; do sleep 1; done
       resources:
         requests:
           cpu: 200m
           memory: 200Mi
         limits:
           cpu: 200m
           memory: 200Mi
       resizePolicy:
       - resourceName: cpu
         restartPolicy: NotRequired
       - resourceName: memory
         restartPolicy: NotRequired
     restartPolicy: Always
   ```

   </Details>
2. While the init container is running, patch the pod with an infeasible resize request:
   ```
   $ kubectl create -f pod.yaml; sleep 5; kubectl patch pod resize-pod --patch '{"spec": {"containers": [{"name": "resize-container", "resources":{"requests": {"cpu": "100"}, "limits": {"cpu": "100"}}}]}}'
   ```
3. Watch the pod:
   ```
   $ kubectl get pod resize-pod -o jsonpath='spec: {.spec.containers[0].resources}{"\nallocatedResources: "}{.status.containerStatuses[0].allocatedResources}{"\nstatus: "}{.status.containerStatuses[0].resources}{"\nresize: "}{.status.resize}{"\n"}' -w
   ```

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
Client Version: v1.30.3
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.30.2
```

</details>

### Cloud provider

N/A

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by hshiina**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
This PR runs the admission process to a pod resize request before passing the request to a pod worker. Otherwise, the pod worker can refer to an unacceptable spec for container life cycle actions such as creating and restarting the container when the resize request is unacceptable.

This fix is inspired by the following comment:
https://github.com/kubernetes/kubernetes/blob/60c4c2b2521fb454ce69dee737e3eb91a25e0535/pkg/kubelet/kubelet.go#L1943-L1951

This PR also uses the latest pod status at resizing like #125958. By copying `apiPodStatus`, which is generated from the latest pod status that a pod worker gets from the PLEG cache, to `pod`, computePodResizeAction() can calculate actions based on the latest pod status.

#### Which issue(s) this PR fixes:

Fixes #126527, Fixes #126033, Fixes  #125205

#### Does this PR introduce a user-facing change?

```release-note
Fixed the following bugs regarding the `InPlacePodVerticalScaling` feature:
- An unacceptable resize is actuated when a container is restarted along with its exit while its `resize` status is stuck in `Deferred` or `Infeasible`.
- An unacceptable resize is actuated if the resize is requested before the container is created.
- It takes one more minute till a pod with the `RestartContainer` resize policy is resized when resize requests to the pod are posted a couple of times quickly.
- A pod status is sometimes regressed to outdated information when resize requests are posted to a pod with the `RestartContainer` resize policy a couple of times quickly.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/e2e_node/pod_resize_test.go [hunk 1]
```diff
@@ -101,38 +103,46 @@ func supportsInPlacePodVerticalScaling(ctx context.Context, f *framework.Framewo
...
 
-func getTestResourceInfo(tcInfo TestContainerInfo) (v1.ResourceRequirements, v1.ResourceList, []v1.ContainerResizePolicy) {
-	var res v1.ResourceRequirements
+func getTestResourceInfo(tcInfo TestContainerInfo) (v1.ResourceRequirements, v1.ResourceRequirements, v1.ResourceList, []v1.ContainerResizePolicy) {
+	var res, statRes v1.ResourceRequirements
 	var alloc v1.ResourceList
 	var resizePol []v1.ContainerResizePolicy
 
-	if tcInfo.Resources != nil {
+	createResourceRequirements := func(resources *ContainerResources) v1.ResourceRequirements {
```
Comment (AnishShah) [inline test/e2e_node/pod_resize_test.go:111]: I refactored pod resize e2e tests in https://github.com/kubernetes/kubernetes/pull/128143. Is it possible to update the test file?

**Comment by hshiina** [comment]
/hold
Using allocated resources for starting a container (issued in #128065) may be the better approach for this issue.

**Comment by tallclair** [comment]
I think https://github.com/kubernetes/kubernetes/pull/128269 supersedes this.

/close

**Pull request closed without merging**